### DISCUSSION TITLE
Invert : Transformer variant group

### DISCUSSION THREAD

**Discussion by lyneal** [unocss/unocss#2612] [Ideas]
Hello,

First of all, thank you for this project which I am a big fan of!
For the moment it is possible to create groups of classes from the variants. 

Example : `sm:(columns-2 bg-red, etc...)`

So I modestly propose this idea/solution to reverse this logic which could be very practical especially for screen variants, it could be easier to read as well as to write from a DX point of view.

Before
`columns-1 sm:columns-2 md:columns-4 lg:columns-3 xl:columns-4 2xl:columns-5`

After
`columns-(1 sm:2 md:4 lg:3 xl:4 2xl:5)`

**Comment by lyneal** [comment]
I haven't had any feedback since I opened this topic. Do you think it's an interesting thought ?

**Reply by Shyam-Chen** [reply]
I didn't use it.

```js
    unocss({
      presets: [presetUno(), presetIcons()],
      transformers: [transformerDirectives({ enforce: 'pre' })],
    }),
```

**Comment by zyyv** [comment]
You can try it in a community transformer.
